### PR TITLE
Refactor accessibility in TimeTool settings and improve tests

### PR DESCRIPTION
**Pull request by OPS-PIvers**
This pull request focuses on improving accessibility and usability of widget settings components, particularly for screen readers and keyboard navigation. The main changes include adding appropriate ARIA attributes, grouping controls semantically, and ensuring labels are properly linked to their controls. There are also minor code cleanups and a `.prettierignore` update.

**Accessibility improvements:**

* Added `role="group"` and `aria-labelledby` to group related controls (e.g., color pickers, preset selectors, mode selectors) in `SurfaceColorSettings.tsx`, `TextSizePresetSettings.tsx`, `TypographySettings.tsx`, and `TimeTool/Settings.tsx` for better screen reader support. Also ensured all interactive elements have proper `aria-pressed` and descriptive `aria-label` attributes. [[1]](diffhunk://#diff-0aaa9149147d88add3b0c5eee09dcaf23754cbf22dc755f3fc313e52693e130dR24-R49) [[2]](diffhunk://#diff-0aaa9149147d88add3b0c5eee09dcaf23754cbf22dc755f3fc313e52693e130dL73-R81) [[3]](diffhunk://#diff-225aef3dafd1f9ab30a07207ed4d77210b5bf6b18cd738e90093e7cef7a012c3L19-R50) [[4]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843R28-R46) [[5]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843L65-R87) [[6]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL81-R96) [[7]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cR124) [[8]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL133-R157) [[9]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cR166) [[10]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL161-R194) [[11]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cR203) [[12]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL220-R263) [[13]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL234-R293) [[14]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cR310) [[15]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL274-R328) [[16]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL285-R350) [[17]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cR366)

* Linked labels to inputs using `useId` and `htmlFor`/`id` in settings components, improving form accessibility for screen readers and keyboard users. [[1]](diffhunk://#diff-0aaa9149147d88add3b0c5eee09dcaf23754cbf22dc755f3fc313e52693e130dR24-R49) [[2]](diffhunk://#diff-225aef3dafd1f9ab30a07207ed4d77210b5bf6b18cd738e90093e7cef7a012c3L19-R50) [[3]](diffhunk://#diff-d28f37536097dd9e4b5ded811b864132c6c47e67ee70964c770f0d570195c843R28-R46) [[4]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL188-R227)

**Code cleanup and refactoring:**

* Removed redundant color name mapping logic from `DraggableWindow.tsx` and ensured color constants are imported consistently. [[1]](diffhunk://#diff-0d33fd1f9e084a22794f65c762da0467ecb5d266454fa1359d74a6156b9c03dcL66-L77) [[2]](diffhunk://#diff-e012e8f9ca853f01f44c6516111545db9c4076f4d3176f325bf40740e0b76c0cL5-R9)

* Replaced inline text size preset logic with a shared utility function for consistency.

**Styling/UI improvements:**

* Adjusted padding and gap logic in the MathTools widget palette for more responsive design.

**Tooling configuration:**

* Updated `.prettierignore` to exclude large, auto-generated markdown tables from formatting, preventing file bloat and tool slowdowns.

### CODE REVIEW

File: components/common/SurfaceColorSettings.tsx [hunk 1]
```diff
@@ -21,24 +21,32 @@ export const SurfaceColorSettings = <
...
+        className="space-y-3 rounded-xl border border-slate-100 bg-slate-50 p-3"
+        role="group"
+        aria-labelledby={surfaceLabelId}
+      >
         <div className="flex flex-wrap gap-2">
           {SURFACE_COLOR_PRESETS.map((color) => (
             <button
               key={color}
               type="button"
+              aria-pressed={cardColor === color}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/SurfaceColorSettings.tsx]: Valid — fixed in d81ca58. Converted every single-selection group from the `aria-pressed` toggle pattern to proper radio semantics: the container is now `role="radiogroup"` and each option is `role="radio"` + `aria-checked`. This covers `SurfaceColorSettings.tsx`, `TextSizePresetSettings.tsx`, both groups in `TypographySettings.tsx` (font family + text color), and all seven single-select groups in `TimeTool/Settings.tsx` (mode, visual type, alert sound, voice level, traffic-light color, clock/number style, theme color). The genuine "Glow" toggle keeps `aria-pressed` as you noted.

For `SurfaceColorSettings` the `radiogroup` role is scoped to the swatch row only (not the outer wrapper), since that wrapper also contains the custom-color `<input>` and the opacity slider, which aren't part of the radio set — it stays `role="group"`.

Note this applies radio *semantics* (announced role + checked state + set position) without adding arrow-key roving-tabindex navigation; each swatch/button remains individually tabbable. That matches the minimal change you suggested and is a clear improvement over the mis-signalled `aria-pressed`. Happy to layer on full arrow-key navigation as a follow-up if desired.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/TimeTool/Settings.tsx [hunk 2]
```diff
@@ -520,6 +599,8 @@ export const TimeToolAppearanceSettings: React.FC<{ widget: WidgetData }> = ({
                     config: { ...config, themeColor: c },
                   })
                 }
+                aria-label={COLOR_HEX_TO_NAME[c] ?? c}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/TimeTool/Settings.tsx]: Valid — fixed in d81ca58. The swatch `aria-label` is now `` `theme color ${COLOR_HEX_TO_NAME[c] ?? c}` ``, mirroring the `DraggableWindow` "annotation color <name>" pattern so screen readers announce a full, contextual label instead of a bare color name.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/TimeTool/Settings.tsx [hunk 3]
```diff
@@ -78,10 +82,18 @@ export const TimeToolSettings: React.FC<{ widget: WidgetData }> = ({
...
           {t('widgets.timeTool.mode')}
         </SettingsLabel>
-        <div className="grid grid-cols-2 gap-2">
+        <div
+          className="grid grid-cols-2 gap-2"
+          role="group"
+          aria-labelledby={`timetool-mode-label-${widget.id}`}
+        >
           {TIME_TOOL_MODES.map((m) => (
             <button
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/TimeTool/Settings.tsx:107]: Valid — fixed in d81ca58. Added explicit `type="button"` to all 12 `<button>` elements in this file (the mode/visual-type/alert-sound/voice-level/traffic-light/clock-style/theme-color options plus the Glow toggle), so the whole file is consistent with the `TypographySettings.tsx` buttons this PR already annotated and there's no latent `type="submit"` hazard if a `<form>` ancestor is ever introduced.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/TimeTool/Settings.tsx [hunk 4]
```diff
@@ -271,7 +332,10 @@ export const TimeToolSettings: React.FC<{ widget: WidgetData }> = ({
         )}
 
         <div className="pt-2 border-t border-slate-100 mt-4">
-          <p className="text-xxs font-bold text-slate-500 uppercase tracking-tight mb-2">
+          <p
+            id={`timetool-trafficlight-label-${widget.id}`}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/TimeTool/Settings.tsx]: Fixed in 0d3a206 — the `id` is now applied to the heading `<p>` only when `hasTrafficLight` is true, so it's present only when the `role="radiogroup"` that references it via `aria-labelledby` is actually rendered. No orphaned labelled element remains when no TrafficLight widget is present. Verified: type-check ✓ lint ✓ tests ✓ (69 TimeTool tests pass).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/TypographySettings.tsx [hunk 5]
```diff
@@ -25,15 +25,26 @@ export const TypographySettings = <
   showColorPicker = true,
 }: TypographySettingsProps<T>) => {
   const { fontFamily = 'global', fontColor = '#334155' } = config;
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/TypographySettings.tsx:27]: No automated fix pushed: the three suggested remedies aren't equivalent — changing the default to `#374151` alters the actual rendered default text color (a visual change), while adding `#334155` to `FONT_COLORS`/`TEXT_COLOR_PRESETS` changes the shared preset palette used by other consumers. Which value is canonical is a product/visual-design decision, so flagging for manual review rather than picking one blindly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/TimeTool/Settings.tsx [hunk 6]
```diff
@@ -78,13 +82,22 @@ export const TimeToolSettings: React.FC<{ widget: WidgetData }> = ({
...
+          icon={TimerIcon}
+        >
           {t('widgets.timeTool.mode')}
         </SettingsLabel>
-        <div className="grid grid-cols-2 gap-2">
+        <div
+          className="grid grid-cols-2 gap-2"
+          role="radiogroup"
+          aria-labelledby={`timetool-mode-label-${widget.id}`}
+        >
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/TimeTool/Settings.tsx:105]: No automated fix pushed: adding roving-tabindex + arrow-key navigation across all 11 radiogroups (ideally via the shared reusable `RadioGroup` wrapper suggested here) is a non-mechanical accessibility feature, and the author already acknowledged it as a deferred follow-up earlier in this PR ("happy to layer on full arrow-key navigation as a follow-up"). Leaving it for that coordinated follow-up rather than shipping a partial unattended change.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/SurfaceColorSettings.tsx [hunk 7]
```diff
@@ -21,24 +21,37 @@ export const SurfaceColorSettings = <
...
+          className="flex flex-wrap gap-2"
+          role="radiogroup"
+          aria-label={`${label} color`}
+        >
           {SURFACE_COLOR_PRESETS.map((color) => (
             <button
               key={color}
               type="button"
+              role="radio"
+              aria-checked={cardColor === color}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/SurfaceColorSettings.tsx:54]: No automated fix pushed: both suggested remedies are structural — adding a synthetic "Custom" swatch vs. moving the `<input type="color">` out of the `role="radiogroup"` container — and each changes component structure/UX differently. Which fits best is a design decision, so flagging for manual review.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-08-11

### Summary
Converts single-select settings groups from the mis-signalled `aria-pressed` toggle pattern to proper radio semantics (`role="radiogroup"` + `role="radio"` + `aria-checked`), adds group-heading labels and explicit `type="button"` across `TimeTool/Settings.tsx` and the shared `SurfaceColorSettings`/`TypographySettings`/`TextSizePresetSettings` components. Clear accessibility improvement over the prior state.

### Code Quality
✓ Consistent radio-semantics pattern; `type="button"` added throughout to avoid latent `type="submit"` hazards.

### Widget Standards
✓ Settings-panel (back-face) changes — normal Tailwind is expected here, no container-query scaling required.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
✓ No impact.

### Regression Risk
Low. One fix pushed this run (commit `0d3a206`): the `timetool-trafficlight-label` `id` now attaches only when its `role="radiogroup"` is rendered (`hasTrafficLight`), removing an orphaned labelled element. Verified type-check ✓ lint ✓ 69 TimeTool tests ✓.

### Missing Tests
Adequate for the shipped semantics. The remaining open threads are triaged, not blockers:
- Roving-tabindex + arrow-key navigation is a known deferred follow-up (needs a shared `RadioGroup`, ideally not hand-rolled per widget).
- Two radiogroup-contract edge cases (default `fontColor` / a custom `cardColor` not in the preset set → no checked member) need a product/design decision on the canonical default.

### Overall
**Merge readiness:** Ready with minor notes — net a11y improvement; the deferred keyboard model and default-checked-member edge cases are tracked in the inline threads.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/SurfaceColorSettings.tsx [hunk 8]
```diff
@@ -21,24 +21,37 @@ export const SurfaceColorSettings = <
...
+      </SettingsLabel>
+      <div
+        className="space-y-3 rounded-xl border border-slate-100 bg-slate-50 p-3"
+        role="group"
+        aria-labelledby={surfaceLabelId}
+      >
+        <div
+          className="flex flex-wrap gap-2"
+          role="radiogroup"
+          aria-label={`${label} color`}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/SurfaceColorSettings.tsx]: Valid — confirmed and fixed in `41d17bc`. `DiceWidget/Settings.tsx` is the only caller passing a label that already names a color (`"Die Color"` at `:55`, `"Pip Color"` at `:64`); the other two call sites pass `"Card surface"` and the `"Surface"` default. So all three composed labels were doubling for Dice exactly as you described.

Rather than dropping the trailing `color` from the template (which would leave `"Card surface"` announcing as just "Card surface", losing the noun) or sanitizing at the call site, the strip happens once in the component so both calling conventions stay valid:

```ts
const colorSubject = label.replace(/\s*colou?rs?\s*$/i, '') || label;
```

Applied to all three sites you flagged — the radiogroup (`:39`), each swatch (`:54`), and the custom-color input (`:66`). Results: `"Die color"` / `"Select die color #ffffff"` / `"Custom die color"`, and unchanged `"Card surface color"` / `"Surface color"` for the other convention. The `|| label` fallback keeps a label of exactly `"Color"` from stripping to an empty accessible name.

Left the opacity slider's `${label} opacity` alone — "Die Color opacity" is accurate (it *is* the opacity of the die color), not a duplication.

New `tests/components/common/SurfaceColorSettings.test.tsx` covers both conventions, the default label, and the exactly-`"Color"` edge case; the doubling test is confirmed to fail against the pre-fix component. Type-check ✓ lint ✓ format ✓, 235 tests green across `tests/components/common` + DiceWidget.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/TypographySettings.tsx [hunk 9]
```diff
@@ -25,15 +25,26 @@ export const TypographySettings = <
...
+          className="grid grid-cols-4 gap-2"
+          role="radiogroup"
+          aria-labelledby={typographyLabelId}
+        >
           {FONTS.map((f) => (
             <button
               key={f.id}
+              type="button"
+              role="radio"
+              aria-checked={fontFamily === f.id}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/TypographySettings.tsx:47]: Valid, and fixed in `41d17bc`. The glyph span now carries `aria-hidden="true"` (`TypographySettings.tsx:63–70`), so each option announces as "School, radio" / "Cursive, radio" rather than folding in the `✏️` / `𝒞` sample character. The `{f.label}` span below still supplies the name, so no `aria-label` was needed on the button. This also matches CLAUDE.md's own baseline ("give purely decorative icons `aria-hidden`").

Added two regression tests (`tests/components/common/TypographySettings.test.tsx`). Worth noting for anyone extending them: they assert on the **computed accessible name**, not `textContent` — `textContent` ignores `aria-hidden`, so a naive version of this test fails even with the fix in place. Both confirmed to fail against the pre-fix component.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/SurfaceColorSettings.tsx [hunk 10]
```diff
@@ -21,24 +21,37 @@ export const SurfaceColorSettings = <
...
+        >
           {SURFACE_COLOR_PRESETS.map((color) => (
             <button
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/SurfaceColorSettings.tsx:50]: Acknowledged, and this remains the standing deferred item rather than a per-PR fix — it's now been raised independently on 2423, 2424, and 2427, so recording the position once more here.

The gap is real: `role="radiogroup"` + `role="radio"` without roving tabindex or arrow-key handling advertises a keyboard contract that isn't implemented. But the scope is the whole codebase, not this diff. `d81ca58` established radiogroup-without-roving-tabindex as the project pattern across `SurfaceColorSettings`, `TextSizePresetSettings`, `TypographySettings`, and the seven `TimeTool/Settings` groups — itself a human decision made in direct response to earlier review feedback on this exact question (the prior round asked for `role="radio"` + `aria-checked` to replace mis-signalled `aria-pressed`, which was the right call and shipped). Retrofitting these four files alone would make the rest of the codebase the outlier without fixing anything.

The right shape is the shared reusable `RadioGroup` wrapper you and the `BuildingSelector` reference both point at, applied in one pass across every consumer — a deliberate change to six-plus files, not something to land unattended mid-review. Logged as backlog; not re-fixing per-instance on further rounds unless the underlying pattern changes.

Two smaller findings from this same round *were* actionable and are fixed in `41d17bc` — the doubled "color" in `SurfaceColorSettings` aria-labels, and the decorative font glyphs leaking into accessible names. See those threads.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/TimeTool/Settings.tsx [hunk 11]
```diff
@@ -78,13 +82,22 @@ export const TimeToolSettings: React.FC<{ widget: WidgetData }> = ({
     <div className="space-y-6 p-1">
       {/* Mode Selection */}
       <div>
-        <SettingsLabel icon={TimerIcon}>
+        <SettingsLabel
+          as="span"
+          id={`timetool-mode-label-${widget.id}`}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/TimeTool/Settings.tsx]: Valid, and applied in `3511623` — all eight `timetool-*-label-${widget.id}` templates are now `useId()` consts, matching the convention this PR introduced in the shared settings primitives. Each id exists once and is referenced twice, so an `id=` / `aria-labelledby=` mismatch is now a compile error rather than a silent break.

Two placement details worth noting:

- `numberStyle` and `colorPalette` turned out to live in `TimeToolAppearanceSettings`, not `TimeToolSettings` — declaring all eight in one component compiled cleanly at first glance but failed `tsc` on the two out-of-scope references, so their `useId()` calls sit in the appearance panel.
- The traffic-light id keeps its conditional spread (`{...(hasTrafficLight ? { id: trafficLightLabelId } : {})}`) from the earlier orphaned-label fix, since its radiogroup only renders when a TrafficLight widget is present.

`grep 'timetool-.*-label'` now returns nothing, no test referenced the literal ids, and behavior is unchanged. Type-check ✓ lint ✓ format ✓, 298 tests green across TimeTool + `tests/components/common`.

---
_Generated by [Claude Code](https://claude.ai/code)_
